### DISCUSSION TITLE
Downsized circuit

### DISCUSSION THREAD

**Discussion by mengstr** [hneemann/Digital#1281] [General]
What?  I just discovered a new representation of the built-in ICs and embedded circuits. Have you announced this somewhere?

<img width="604" alt="Screenshot 2024-04-21 at 11 39 57" src="https://github.com/hneemann/Digital/assets/325326/472fee2e-89ff-45be-afcb-c8dd98992d2b">

**Comment by mengstr** [comment]
I see now that it's probably a feature under development in the pre-release I'm using....  The '138 IC is not rendering 100% correctly.  But it's a nice idea/feature - especially if it could show the internal state every now and then and not be a static image.

<img width="675" alt="Screenshot 2024-04-21 at 11 49 43" src="https://github.com/hneemann/Digital/assets/325326/415a2200-2e77-465c-ba38-281ddeaf3159">

**Reply by hneemann** [reply]
The function is fully implemented. The inner circuit is scaled to half size. This means that pins can fall between grid points. As these cannot then be contacted, the pins are moved to the next full grid point. In order for the inner circuit to be displayed correctly, the pins in the inner circuit must omit every second grid point.
Therefore, not every circuit is suitable for this type of shape.

**Reply by mengstr** [reply]
Ahhh.... Now when you explain it it's totally obvious. I really should have realised that myself. :-)
